text/0000-thread-affinity.md

```diff
@@ -0,0 +1,133 @@
+- Feature Name: thread_affinity
+- Start Date: 2016-01-26
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Thread affinity provides functionality to lock/unlock a specific thread to a specified CPU core.
+The locked threads are then not scheduled so they do not ever move to other cores by the OS scheduler. I believe it useful for a systems programmer to have the ability to set/unset set thread affinity for Rust threads.
+
+# Motivation
+[motivation]: #motivation
+
+My motivation derives from a personal use-case of creating a fiber-based server/client application framework for Rust in my spare time. I need the ability to set thread affinity of Rust threads in an OS-indepedent manner to CPU cores to minimize cache misses and thread context switching.
+
+The proposed feature provides a performance optimization for servers. It would also be useful for MIO and similar frameworks where the main event-loop is locked to one core and passes connections to workers that are also locked to the remaining cores so it takes advantage of cache-locality.
+
+# Detailed design
+[design]: #detailed-design
+
+I strongly encourage others to improve my RFC since I am not a Rust expert and only serves to start a dialog on a proposed API.
+
+I split it into two parts: *User API* and *Implementation*.
+
+### Locking to an arbitrary free CPU core
```

Comment:
There's a lot of room for detail on how exactly it locks to an arbitrary free CPU core. Does it determine what CPU core the thread is currently executing on and then lock to that? Does it pick a random CPU core?